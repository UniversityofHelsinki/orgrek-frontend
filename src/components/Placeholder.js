import Typography from '@mui/material/Typography';
import React from 'react';

/**
 * Displays a placeholder text if empty is true, otherwise the content.
 *
 * @param empty
 * @param placeholder placeholder text to display
 * @param align text-align of the placeholder
 * @param children displayed if empty is false
 * @param props other props passed to placeholder Typography
 */
const Placeholder = ({
  empty,
  placeholder,
  align = 'center',
  children,
  ...props
}) => {
  if (!empty) {
    return children;
  }

  return (
    <Typography variant="body1" color="text.secondary" align={align} {...props}>
      {placeholder}
    </Typography>
  );
};

export default Placeholder;
