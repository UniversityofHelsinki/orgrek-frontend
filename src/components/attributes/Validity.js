import React from 'react';
import { useTranslation } from 'react-i18next';
import { showValidity } from '../../actions/utilAction';

const Validity = ({ startDate, endDate }) => {
  const { t, i18n } = useTranslation();

  return (
    <span data-testid="validity">
      {showValidity(startDate, endDate, i18n, t)}
    </span>
  );
};

export default Validity;
