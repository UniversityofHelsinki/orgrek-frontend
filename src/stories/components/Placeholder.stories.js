import React from 'react';
import Placeholder from '../../components/Placeholder';
import Icon from '@mui/icons-material/ErrorOutline';
import Validity from '../../components/attributes/Validity';

export default {
  component: Placeholder,
  argTypes: {
    align: {
      control: 'radio',
      options: ['left', 'center', 'right', 'justify'],
    },
  },
};

export const TextPlaceholder = {
  args: {
    empty: true,
    placeholder: 'Placeholder text',
    align: 'center',
  },
  render: (args) => <Placeholder {...args}>Content if not empty</Placeholder>,
};

export const ComponentPlaceholder = {
  args: {
    empty: true,
    align: 'center',
  },
  render: (args) => (
    <Placeholder {...args} placeholder={<Icon />}>
      Content if not empty
    </Placeholder>
  ),
};

export const ValidityPlaceholder = {
  args: {
    empty: true,
    align: 'center',
  },
  render: (args) => (
    <Placeholder {...args} placeholder={<Validity />}>
      <Validity startDate="2023-01-01" />
    </Placeholder>
  ),
};
