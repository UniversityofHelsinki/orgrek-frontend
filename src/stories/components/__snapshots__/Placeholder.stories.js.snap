// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/Placeholder ComponentPlaceholder smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": undefined,
  "name": "stories-components-placeholder--component-placeholder",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/Placeholder TextPlaceholder smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Placeholder text",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-placeholder--text-placeholder",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/Placeholder ValidityPlaceholder smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "not_specified",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-placeholder--validity-placeholder",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
